src/Sil/Bundle/ProductBundle/Resources/config/services.yml

```diff
@@ -153,6 +205,23 @@ services:
             - [setAttributeTypeClass, ['%sil.model.product_attribute_type.class%']]
             - [setAttributeTypeRepository, ['@sil.repository.product_attribute_type']]
 
+    sil_product.form.transformer.array_to_option_transformer:
+        class: Sil\Bundle\ProductBundle\Form\Transformer\ArrayToOptionTransformer
+        calls:
+            - [setOptionClass, ['%sil.model.product_option.class%']]
```

Comment:
Why not directly inject as constructor argument ?

Reply:
Because `transform` method does not use any injected service, so it can be lazily injected when `reverseTransform` only.